testcases.py

```diff
@@ -736,6 +743,97 @@ def scenario() -> str:
         return "corrupt-rate --delay=15ms --bandwidth=10Mbps --queue=25 --rate_to_server=2 --rate_to_client=2"
 
 
+class TestCaseECN(TestCaseHandshake):
+    @staticmethod
+    def name():
+        return "ecn"
+
+    @staticmethod
+    def abbreviation():
+        return "E"
+
+    @staticmethod
+    def testname(p: Perspective):
+        return "handshake"
+
+    def count_ecn(self, tr):
+        ecn = [0] * (max(ECN) + 1)
+        for p in tr:
+            e = int(getattr(p["ip"], "dsfield.ecn"))
```

Comment:
What exactly does this do?

Reply:
It counts the number of packets marked with each distinct ECN codepoint.